components/common/src/main/java/com/hotels/styx/common/format/SanitisingThrowableInterceptor.java

```diff
@@ -0,0 +1,103 @@
+/*
+  Copyright (C) 2013-2020 Expedia Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx.common.format;
+
+import net.bytebuddy.implementation.bind.annotation.AllArguments;
+import net.bytebuddy.implementation.bind.annotation.Origin;
+import net.bytebuddy.implementation.bind.annotation.RuntimeType;
+
+import java.lang.reflect.Method;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Intercepts methods of a {@link Throwable} so that the message can be modified to sanitize the values of any recognised cookies.
+ * Any cause is similarly wrapped before being returned.
+ */
+public class SanitisingThrowableInterceptor {
```

Comment:
This is a matter of taste I guess, but I think we are exposing too many details with this SanitisingThrowableFactory + SanitisingThrowableInterceptor . Probably I would try to have both classes in the same file  and use simpler names.

Reply:
Thanks - yes, a bit of a rethink, and I agree it looks much better.